src/lib.rs

```diff
@@ -1247,8 +1247,26 @@ impl<'a> MatchTreeStep<'a> {
             Format::Bits(_a) => {
                 Self::accept() // FIXME
             }
-            Format::WithRelativeOffset(_expr, _a) => {
-                Self::accept() // FIXME
+            Format::WithRelativeOffset(expr, a) => {
+                // REVIEW - this is a hack but it is at least somewhat better than before
+                let tree = Self::add_next(module, next.clone());
+                let bounds = expr.bounds();
+                let peek = match bounds.is_exact() {
+                    Some(0) => Self::add(module, a, Rc::new(Next::Empty)),
+                    Some(n) => Self::add_slice(
+                        module,
+                        n,
+                        Rc::new(Next::Empty),
+                        Rc::new(Next::Tuple(std::slice::from_ref(a.as_ref()), next)),
+                    ),
+                    None => Self::add_slice(
+                        module,
+                        bounds.min,
+                        Rc::new(Next::Empty),
+                        Rc::new(Next::Empty),
+                    ),
```

Comment:
Or at least this part is unlikely to be necessary.

Reply:
Ah, I see. I can narrow the peek to just the cncrete case, and otherwise just default to the value `tree`. Does that work?

Reply:
I changed it accordingly, let me know if there are still any points to revise.